src/ci/docker/host-x86_64/mingw-check-tidy/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:22.04
+FROM ghcr.io/rust-lang/ubuntu:22.04
```

Comment:
Might be nice to add a comment here to explain why do we use an image from ghcr, and why we do it for this specific CI job.